src/stores/userImage.ts

```diff
@@ -0,0 +1,354 @@
+      }
+      // フロント側でも追加
+      this.uploadedImages = [...this.uploadedImages, newImage]
+      try {
+        const docRef = doc(db, 'userImageStorage', uid.value)
+        const userImages = await this.getImageDocSnap(docRef)
+
+        if (userImages === undefined) {
+          const newUserImages = {
+            [id]: newImage,
+          }
+          await setDoc(docRef, {
+            images: newUserImages,
+          })
+        } else {
+          userImages[id] = newImage
+          await updateDoc(docRef, {
+            images: userImages,
+          })
+        }
+      } catch (e) {
+        console.log('Error getting cached document:', e)
+      }
+    },
+
+    async setUserImageToToolbar() {
+      const { uid } = storeToRefs(useAuthStore())
+      const docRef = doc(db, 'userImageStorage', uid.value)
+
+      const userImages = await this.getImageDocSnap(docRef)
+      // ツールバーに画像をセット if(show === true)
+      if (userImages !== undefined) {
+        Object.keys(userImages).forEach((imageId) => {
+          const userImage: UploadedImage = userImages[imageId]
+          if (userImage.show) {
+            this.uploadedImages.push(userImage)
+          }
+        })
+
+        // アップロード順に並び替える
+        const sortedImages = this.uploadedImages.sort(
+          (a: UploadedImage, b: UploadedImage) => {
+            if (a.createdAt < b.createdAt) {
+              return -1
+            }
+            if (a.createdAt > b.createdAt) {
+              return 1
+            }
+            return 0
+          },
+        )
+        this.uploadedImages = sortedImages
+        this.isLoadingImages = false
+      } else {
+        this.isLoadingImages = false
+      }
+    },
+
+    // ツールバーから画像を削除
+    async deleteImageFromToolbar(image: UploadedImage) {
+      // フロント側で削除
+      this.uploadedImages = this.uploadedImages.filter((i) => i.id !== image.id)
+      // firestore上でshowをfalseにする
+      const docRef = doc(db, 'userImageStorage', image.userUid)
+
+      const userImages = await this.getImageDocSnap(docRef)
+
+      if (userImages !== undefined) {
+        const userImage = userImages[image.id]
+        userImage.show = false
+        await updateDoc(docRef, {
+          images: userImages,
+        })
+      }
+    },
+
+    // Storage削除判定
+    canDeleteImageFromStorage(uploadedImage: UploadedImage) {
+      // 条件canvasで使用されていないかつToolbarで非表示
+      return (
+        Object.keys(uploadedImage.canvases).length === 0 && !uploadedImage.show
+      )
+    },
+
+    // ログイン時に呼び出し、削除判定をして画像削除
+    async deleteImageFromStorageWithLogin() {
+      const { uid } = storeToRefs(useAuthStore())
+      const docRef = doc(db, 'userImageStorage', uid.value)
+      const userImages = await this.getImageDocSnap(docRef)
+      if (userImages !== undefined) {
+        Object.keys(userImages).forEach((imageId) => {
+          const userImage: UploadedImage = userImages[imageId]
+          if (this.canDeleteImageFromStorage(userImage)) {
```

Comment:
```suggestion
           const isDeleteImageFromStorage = (...)
           if (isDeleteImageFromStorage) {
```

長い条件を分けていて良いですね！
if判定は変数に入れるだけで良いと思います。
宗教かもですが、`can`でなく`is`が一般的に使われてます